internal/handlers/handler_verify.go

```diff
@@ -235,66 +253,174 @@ func updateActivityTimestamp(ctx *middlewares.AutheliaCtx, isBasicAuth bool, use
+		ctx.Providers.Authorizer.IsURLMatchingRuleWithGroupSubjects(*targetURL) &&
+		(refreshProfileInterval == schema.RefreshIntervalAlways || userSession.RefreshTTL.Before(ctx.Clock.Now())) {
+		ctx.Logger.Debugf("Checking the authentication backend for an updated profile for user %s", userSession.Username)
+		details, err := ctx.Providers.UserProvider.GetDetails(userSession.Username)
+		// Only update the session if we could get the new details.
+		if err != nil {
+			return err
+		}
+
+		groupsDiff := utils.IsStringSlicesDifferent(userSession.Groups, details.Groups)
+		emailsDiff := utils.IsStringSlicesDifferent(userSession.Emails, details.Emails)
+		if !groupsDiff && !emailsDiff {
+			// TODO: FOR 4.16.1 RELEASE change Debugf to Tracef
+			ctx.Logger.Debugf("Updated profile not detected for %s.", userSession.Username)
+		} else {
+			ctx.Logger.Debugf("Updated profile detected for %s.", userSession.Username)
+			// TODO: FOR 4.16.1 RELEASE: change "debug" to "trace"
+			if ctx.Configuration.LogLevel == "debug" {
+				generateVerifySessionHasUpToDateProfileTraceLogs(ctx, userSession, details)
+			}
+			userSession.Groups = details.Groups
+			userSession.Emails = details.Emails
+
+			// Only update TTL if the user has a interval set.
+			if refreshProfileInterval != schema.RefreshIntervalAlways {
+				userSession.RefreshTTL = ctx.Clock.Now().Add(refreshProfileInterval)
+				return ctx.SaveSession(*userSession)
+			}
+		}
+		// Only update TTL if the user has a interval set.
+		// Also make sure to update the session even if no difference was found.
+		// This is so that we don't check every subsequent request after this one.
+		if refreshProfileInterval != schema.RefreshIntervalAlways {
+			userSession.RefreshTTL = ctx.Clock.Now().Add(refreshProfileInterval)
+			return ctx.SaveSession(*userSession)
+		}
 	}
+	return nil
+}
 
-	if !isURLUnderProtectedDomain(targetURL, ctx.Configuration.Session.Domain) {
-		ctx.Logger.Error(fmt.Errorf("The target URL %s is not under the protected domain %s",
-			targetURL.String(), ctx.Configuration.Session.Domain))
-		ctx.ReplyUnauthorized()
-		return
+func getProfileRefreshSettings(cfg schema.Configuration) (refresh bool, refreshInterval time.Duration) {
+	if cfg.AuthenticationBackend.Ldap != nil {
+		if cfg.AuthenticationBackend.RefreshInterval != schema.ProfileRefreshDisabled {
+			refresh = true
+			if cfg.AuthenticationBackend.RefreshInterval != schema.ProfileRefreshAlways {
+				// Skip Error Check since validator checks it
+				refreshInterval, _ = utils.ParseDurationString(cfg.AuthenticationBackend.RefreshInterval)
+			} else {
+				refreshInterval = schema.RefreshIntervalAlways
+			}
+		}
 	}
+	return refresh, refreshInterval
+}
 
-	var username string
-	var groups []string
-	var authLevel authentication.Level
+// VerifyGet returns the handler verifying if a request is allowed to go through.
+func VerifyGet(cfg schema.Configuration) middlewares.RequestHandler {
```

Comment:
I think you can pass only what you need. schema.AuthenticationBackendConfiguration is sufficient. Limiting the scope It helps figure out what the function is likely to use from the call location.